### PR TITLE
add other liabilities section to ixbrl balance sheet

### PR DESCRIPTION
**Pull request by leahtaylor**
Add other liabilities section to ixbrl balance sheet (current and previous period)

Prepayments and accrued income 
Creditors - amounts falling due within one year 
Net current assets (liabilities): 
Total assets less current liabilities: 
Creditors - amounts falling due after more than one year:
Provision for liabilities: 
Accruals and deferred income: 
Permitted characters (all fields):
Total net assets: 

SFA-248, SFA-472

### CODE REVIEW

**Comment by leahtaylor** [comment]
screen shot 2018-10-22 at 14 58 46
screen shot 2018-10-22 at 14 54 52

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 1]
```diff
@@ -1738,100 +1736,302 @@ <h2 class="text--center">Balance sheet</h2>
...
 
-                <tr>
-                  <th class = "strong" id="current-assets-heading">Current assets</th>
-                </tr>
+                    <tr>
+                      <th class = "strong" id="current-assets-heading">Current assets</th>
+                    </tr>
 
-                <!-- BALANCE SHEET - CURRENT ASSETS - STOCKS -->
+                    <!-- BALANCE SHEET - CURRENT ASSETS - STOCKS -->
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Can we re-align from line 1739-1744?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 2]
```diff
@@ -1768,7 +1768,7 @@ <h2 class="text--center">Balance sheet</h2>
 
               {{if $debtors := $currentAssets.debtors}}
                 <tr class="figures">
-                  <td id="ac52-label">Debtors:</td>
+                  <td id="debtors-label">Debtors:</td>
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Do we need to match our ids to the ids in the provided SmallFullAccountsModifiediXBRL.html?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 3]
```diff
@@ -1830,6 +1830,189 @@ <h2 class="text--center">Balance sheet</h2>
...
               {{end}}
+              {{end}}
+
+              <!-- BALANCE SHEET - LIABILITIES -->
+
+              {{if $liabilities := $balanceSheet.other_liabilities_and_assets}}
+
+                  {{if $prepayments := $liabilities.prepayments_and_accrued_income}}
+                      <tr class="figures">
+                          <td id="prepayments-label">Prepayments and accrued income:</td>
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Same here. 
Do we need to match our ids to the ids in the provided SmallFullAccountsModifiediXBRL.html?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 4]
```diff
@@ -1830,6 +1830,189 @@ <h2 class="text--center">Balance sheet</h2>
...
+
+                  {{if $creditors_within := $liabilities.creditors_amounts_falling_due_within_one_year}}
+                      <tr class="figures">
+                          <td id="creditors-within-label">Creditors: amounts falling due within one year:</td>
+
+                          <td class="figure noteIndex" id="uk-busCreditorsCreditorsWithinOneYear_CY_ENDNoteIndexes"></td>
+
+                          <td id="creditors-within-curr-val" class="strong figure">
+                              {{if isNotEmpty $creditors_within.current_amount}}
+                                  {{if gt $creditors_within.current_amount 0.0}}({{end}}
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Do we use parentheses for positive amounts?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 5]
```diff
@@ -1830,6 +1830,189 @@ <h2 class="text--center">Balance sheet</h2>
...
+                                  <ix:nonFraction contextRef="CreditorsWithinOneYear_CY_END" decimals="0" unitRef="GBP" name="core:Creditors" format="ixt2:numdotdecimal">{{$creditors_within.current_amount}}</ix:nonFraction>
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: We could indent this line.

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 6]
```diff
@@ -1830,6 +1830,189 @@ <h2 class="text--center">Balance sheet</h2>
...
+                                  {{if isNotEmpty $creditors_after.previous_amount}}
+                                  {{if gt $creditors_after.previous_amount 0.0}}({{end}}
+                                  <ix:nonFraction contextRef="CreditorsAfterOneYear_PY_END" decimals="0" unitRef="GBP" name="core:Creditors" format="ixt2:numdotdecimal">{{$creditors_after.previous_amount}}</ix:nonFraction>
+                                  {{if gt $creditors_after.previous_amount 0.0}}){{end}}
+                                  {{end}}
+                              </td>
+                          </tr>
+                      {{end}}
+
+                      {{if $provision := $liabilities.provision_for_liabilities}}
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Can we align this ` {{if $provision := $liabilities.provision_for_liabilities}}` block with the beginning of the the previous block `{{if $creditors_after := $liabilities.creditors_amounts_falling_due_after_more_than_one_year}}`?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 7]
```diff
@@ -1830,6 +1830,189 @@ <h2 class="text--center">Balance sheet</h2>
...
+                      <tr class="figures">
+                          <td id="provision-for-liabilities-label">Provision for liabilities:</td>
+
+                          <td class="figure noteIndex" id="uk-busProvisionsForLiabilitiesBalanceSheetSubtotalCY_ENDNoteIndexes"></td>
+
+                          <td id="provision-curr-val" class="strong figure">
+                              {{if isNotEmpty $provision.current_amount}}
+                              {{if gt $provision.current_amount 0.0}}({{end}}
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Do we use parentheses for positive amounts?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 8]
```diff
@@ -1830,6 +1830,189 @@ <h2 class="text--center">Balance sheet</h2>
...
+                              <ix:nonFraction contextRef="CY_END" decimals="0" unitRef="GBP" name="core:ProvisionsForLiabilitiesBalanceSheetSubtotal" format="ixt2:numdotdecimal">{{$provision.current_amount}}</ix:nonFraction>
+                              {{if gt $provision.current_amount 0.0}}){{end}}
+                              {{end}}
+                          </td>
+
+                          <td id="provision-prev-val" class="strong figure">
+                              {{if isNotEmpty $provision.previous_amount}}
+                              {{if gt $provision.previous_amount 0.0}}({{end}}
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Same here: Do we use parentheses for positive amounts?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 9]
```diff
@@ -1830,6 +1830,189 @@ <h2 class="text--center">Balance sheet</h2>
...
+                              <ix:nonFraction contextRef="PY_END" decimals="0" unitRef="GBP" name="core:ProvisionsForLiabilitiesBalanceSheetSubtotal" format="ixt2:numdotdecimal">{{$provision.previous_amount}}</ix:nonFraction>
+                              {{if gt $provision.previous_amount 0.0}}){{end}}
+                              {{end}}
+                          </td>
+                      </tr>
+                      {{end}}
+
+                      {{if $accruals := $liabilities.accruals_and_deferred_income}}
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Can we align this block?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 10]
```diff
@@ -1830,6 +1830,189 @@ <h2 class="text--center">Balance sheet</h2>
...
+                      <tr class="figures">
+                          <td id="accruals-and-deferred-income-label">Accruals and deferred income:</td>
+
+                          <td class="figure noteIndex" id="uk-busAccruedLiabilitiesNotExpressedWithinCreditorsSubtotalCY_ENDNoteIndexes"></td>
+
+                          <td id="accruals-curr-val" class="strong figure">
+                              {{if isNotEmpty $accruals.current_amount}}
+                              {{if gt $accruals.current_amount 0.0}}({{end}}
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Same, parentheses for positive amount?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 11]
```diff
@@ -1830,6 +1830,189 @@ <h2 class="text--center">Balance sheet</h2>
...
+                              <ix:nonFraction contextRef="CY_END" decimals="0" unitRef="GBP" name="core:AccruedLiabilitiesNotExpressedWithinCreditorsSubtotal" format="ixt2:numdotdecimal">{{$accruals.current_amount}}</ix:nonFraction>
+                              {{if gt $accruals.current_amount 0.0}}){{end}}
+                              {{end}}
+                          </td>
+
+                          <td id="accruals-prev-val" class="strong figure">
+                              {{if isNotEmpty $accruals.previous_amount}}
+                              {{if gt $accruals.previous_amount 0.0}}({{end}}
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Same, parentheses for positive amount?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 12]
```diff
@@ -1738,100 +1736,302 @@ <h2 class="text--center">Balance sheet</h2>
...
 
-                <!-- BALANCE SHEET - CURRENT ASSETS - DEBTORS -->
+              <!-- BALANCE SHEET - CURRENT ASSETS - DEBTORS -->
 
-              {{if $debtors := $currentAssets.debtors}}
-                <tr class="figures">
-                  <td id="ac52-label">Debtors:</td>
+                  {{if $debtors := $currentAssets.debtors}}
+                    <tr class="figures">
+                      <td id="debtors-label">Debtors:</td>
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Do we need to match our ids to the ids in the provided SmallFullAccountsModifiediXBRL.html?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 13]
```diff
@@ -1738,100 +1736,302 @@ <h2 class="text--center">Balance sheet</h2>
...
+
+                      {{if $provision := $liabilities.provision_for_liabilities}}
+                          <tr class="figures">
+                              <td id="provision-for-liabilities-label">Provision for liabilities:</td>
+
+                              <td class="figure noteIndex" id="uk-busProvisionsForLiabilitiesBalanceSheetSubtotalCY_ENDNoteIndexes"></td>
+
+                              <td id="provision-curr-val" class="strong figure">
+                                  {{if isNotEmpty $provision.current_amount}}
+                                  {{if gt $provision.current_amount 0.0}}({{end}}
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Do we use parentheses for positive amounts?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 14]
```diff
@@ -1738,100 +1736,302 @@ <h2 class="text--center">Balance sheet</h2>
...
+                                  <ix:nonFraction contextRef="CY_END" decimals="0" unitRef="GBP" name="core:ProvisionsForLiabilitiesBalanceSheetSubtotal" format="ixt2:numdotdecimal">{{$provision.current_amount}}</ix:nonFraction>
+                                  {{if gt $provision.current_amount 0.0}}){{end}}
+                                  {{end}}
+                              </td>
+
+                              <td id="provision-prev-val" class="figure">
+                                  {{if isNotEmpty $provision.previous_amount}}
+                                  {{if gt $provision.previous_amount 0.0}}({{end}}
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Same here: Do we use parentheses for positive amounts?

File: src/main/resources/filing/templates/small-full-accounts.html [hunk 15]
```diff
@@ -1738,100 +1736,302 @@ <h2 class="text--center">Balance sheet</h2>
...
+                              {{if isNotEmpty $previous_total_assets}}
+                                   <ix:nonFraction contextRef="PY_END" decimals="0" unitRef="GBP" name="core:TotalAssetsLessCurrentLiabilities" format="ixt2:numdotdecimal">{{$previous_total_assets}}</ix:nonFraction>
+                              {{end}}
+                            </td>
+                        </tr>
+                  {{end}}
+
+              <!-- BALANCE SHEET - LIABILITIES - CREDITORS AFTER ONE YEAR -->
+
+                     {{if $creditors_after := $liabilities.creditors_amounts_falling_due_after_more_than_one_year}}
```
Comment (ch-mcabrera) [inline src/main/resources/filing/templates/small-full-accounts.html]: Can we align this `{{if $creditors_after := $liabilities.creditors_amounts_falling_due_after_more_than_one_year}}` block 
with the beginning of the the previous block `{{if isNotEmpty ($current_total_assets := $liabilities.current_total_assets_less_current_liabilities)`?

File: src/main/resources/filing/templates/small-full-accounts.html (hunk 13 above)
Comment (leahtaylor) [reply src/main/resources/filing/templates/small-full-accounts.html]: These values are subtracted from the total, so any value over 0 is in brackets to show it is a deduction.

**Comment by leahtaylor** [comment]
@ch-mcabrera carl palfrey has said that the naming of the ids is not important and do not need to necessarily match the prototype. Have made the naming consistent throughout the template

File: src/main/resources/filing/templates/small-full-accounts.html (hunk 13 above)
Comment (ch-mcabrera) [reply src/main/resources/filing/templates/small-full-accounts.html]: Perfect. I couldn't remember the logic and there was nothing in the story.

**Comment by ch-mcabrera** [comment]
@leahtaylor 
I am happy with the latest changes. I can see the point of the formatting (`getMagnitudeFormatted()`) inside the ixbrl. 
But we could check with the lead dev if they are happy with us to do so.

**Pull request merged**